libs/api/src/main/java/de/cyface/api/MeasurementIterator.java

```diff
@@ -173,8 +173,10 @@ public void handle(final JsonObject document) {
     public void load(final Handler<Measurement> writeHandler, final Handler<Void> separationHandler,
             final Handler<Void> endHandler, final Handler<Throwable> failureHandler) {
 
-        final var hasNext = hasNext();
-        if (hasNext) {
+        if (!initialized) {
+            // Seems like no entries where found for the requested data
+            endHandler.handle(null);
+        } else if (hasNext()) {
```

Comment:
This fixes a bug: When the `MeasurementIterator` was called for data which has no buckets in the database, this class never resolved. Now it calls the `initializedHandler` and returns the iterator with hasNext = false.